### ISSUE TITLE
[BUG] No such file or directory, open 'node_modules\src\util\root.ts'

### ISSUE THREAD

**Issue by andrew19881123** [ng-packagr/ng-packagr#758]
## Type of Issue

```
[x ] Bug Report
[ ] Feature Request
```

## Description
```
> ng-packagr -p package.json

Building Angular library
Generating bundle for project
Cleaning bundle build directory
Processing assets
Running ngc
Compiling to FESM15
'default' is not exported by 'node_modules\rxjs\add\operator\toPromise.js'
Compiling to FESM5
Compiling to UMD
Minifying UMD bundle
Side effects in initialization of unused variable ez [0:80,12]
Dropping side-effect-free statement [0:2738,17]
Dropping side-effect-free statement [0:3243,21]
Dropping side-effect-free statement [0:4926,17]
Dropping side-effect-free statement [0:6090,21]
Dropping duplicated definition of variable gV [0:6518,16]
Dropping duplicated definition of variable fV [0:6525,17]
Dropping unreachable code [0:15959,0]
Declarations in unreachable code! [0:15959,0]
Dropping unreachable code [0:17346,0]
Declarations in unreachable code! [0:17346,0]
Side effects in initialization of unused variable sya [0:19408,53]
Side effects in initialization of unused variable tya [0:19408,89]
Side effects in initialization of unused variable uya [0:19408,131]
Dropping duplicated definition of variable nFa [0:21288,17]
Side effects in initialization of unused variable V [0:813,4]
Side effects in initialization of unused variable W [0:814,4]
Side effects in initialization of unused variable fa [0:1128,4]
Side effects in initialization of unused variable ga [0:1129,4]
Side effects in initialization of unused variable ha [0:1130,4]
Dropping duplicated definition of variable f [0:1537,4]
Side effects in initialization of unused variable Ab [0:2648,4]
Side effects in initialization of unused variable Bb [0:2649,4]
Side effects in initialization of unused variable Cb [0:2650,4]
Side effects in initialization of unused variable Ip [0:15332,4]
Side effects in initialization of unused variable ev [0:19436,4]
Side effects in initialization of unused variable fv [0:19437,4]
Side effects in initialization of unused variable $x [0:21698,4]
Side effects in initialization of unused variable ly [0:22074,4]
Side effects in initialization of unused variable my [0:22075,4]

BUILD ERROR
ENOENT: no such file or directory, open 'C:\Users\xxxxxx\node_modules\src\util\root.ts'
Error: ENOENT: no such file or directory, open 'C:\Users\xxxxxx\node_modules\src\util\root.ts'

npm ERR! code ELIFECYCLE
npm ERR! errno 111
npm ERR! xxx@1.0.0 packagr: `ng-packagr -p package.json`
npm ERR! Exit status 111
npm ERR!
npm ERR! Failed at the xxx@1.0.0 packagr script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\xx\AppData\Roaming\npm-cache\_logs\2018-04-09T11_24_23_994Z-debug.log
```

#### Version Information
npm : 5.6.0
node: 8.9.4
```
+-- @angular/cli@1.4.10
+-- @angular/common@4.4.6
+-- @angular/compiler@4.4.6
+-- @angular/compiler-cli@4.4.6
+-- @angular/core@4.4.6
+-- @angular/forms@4.4.6
+-- @angular/http@4.4.6
+-- @angular/language-service@4.4.6
+-- @angular/platform-browser@4.4.6
+-- @angular/platform-browser-dynamic@4.4.6
+-- @angular/router@4.4.6
+-- @compodoc/compodoc@1.0.6
+-- @types/jasmine@2.5.54
+-- @types/jasminewd2@2.0.3
+-- @types/node@6.0.104
+-- codelyzer@3.1.2
+-- core-js@2.5.5
+-- jasmine-core@2.6.4
+-- jasmine-spec-reporter@4.1.1
+-- karma@1.7.1
+-- karma-chrome-launcher@2.1.1
+-- karma-cli@1.0.1
+-- karma-coverage-istanbul-reporter@1.2.1
+-- karma-jasmine@1.1.1
+-- karma-jasmine-html-reporter@0.2.2
+-- ng-packagr@1.7.0
+-- protractor@5.3.0
+-- rollup-plugin-commonjs@8.3.0
+-- rxjs@5.5.8
+-- ts-node@3.2.2
+-- tslint@5.3.2
+-- typescript@2.3.4
`-- zone.js@0.8.26
```

**Comment by YuriGP** [comment]
I'm having the same problem. It was working few days ago so probably it's a ng-packagr dependency.

**Comment by milocosmopolitan** [comment]
same here. it just this bug started to occur since this morning

**Comment by ninjadq** [comment]
I'm having the same problem today.

**Comment by dmeneses** [comment]
yeap, same here

**Comment by knlreddy** [comment]
Me too....I am getting the same error for 'isFunction.ts' file as below. Please update if any one has found the fix to resolve this issue.

BUILD ERROR
ENOENT: no such file or directory, open '<SourceRoot>\node_modules\src\util\isFunction.ts'
Error: ENOENT: no such file or directory, open '<SourceRoot>\node_modules\src\util\isFunction.ts'

**Comment by dmeneses** [comment]
Seems that the culprit is: uglify-js. I run:

`npm install uglify-js@3.3.18`

and try to build and worked.

**Comment by knlreddy** [comment]
Thank you, dmeneses. It worked for me. I ran the npm install command have you given and the code started building now...no issues. Appreciate your help, thank you again!!

**Comment by ninjadq** [comment]
I tried `npm install uglify-js@3.3.18 `, but the problem seems still exists.

**Comment by ninjadq** [comment]
my current version is 1.6.x, when I upgrade to 1.7.x and degrade the uglify-js, it works, thank you @dmeneses

**Comment by kmikodev** [comment]
@ninjadq I tried your solution and it works thanks

**Comment by ghiscoding** [comment]
Hey guys, I also encountered this problem while running an `npm install` on my lib project. This was fixed in latest `2.x` version but if you are running `1.x`, then you could edit a file under the `node_modules/ng-packagr/src/lib/flatten/uglify.js` and do this change
```diff
const result = minify(inputFileBuffer.toString(), {
      sourceMap: {
+      includeSources: true,
        content: JSON.parse(inputSourceMapBuffer.toString()),
        url: basename(sourcemapOut)
      },
```
as can be seen in the `2.x version` [commit](https://github.com/dherges/ng-packagr/commit/ff176b74fc2e425a5d8846f89f76dcd797cd9211) fix that was merged and released under `2.4.2`

I think this will work with any version of `uglify-js` that is `>=3.x`

**Comment by SubinY** [comment]
It`s amazing. My project did not depend the package uglify-js. I think this question in regard to the version of ng-packagr, so i update the version to 1.7x like @ninjadq said, but it does not work, when i install uqlify-js, the component package is been packed successfully. thank the guys upstairs.

**Comment by venkatesh126** [comment]
Thank you so its working fine...

**Comment by CarreraPHP** [comment]
> Hey guys, I also encountered this problem while running an `npm install` on my lib project. This was fixed in latest `2.x` version but if you are running `1.x`, then you could edit a file under the `node_modules/ng-packagr/src/lib/flatten/uglify.js` and do this change
> 
> ```diff
> const result = minify(inputFileBuffer.toString(), {
>       sourceMap: {
> +      includeSources: true,
>         content: JSON.parse(inputSourceMapBuffer.toString()),
>         url: basename(sourcemapOut)
>       },
> ```
> 
> as can be seen in the `2.x version` [commit](https://github.com/dherges/ng-packagr/commit/ff176b74fc2e425a5d8846f89f76dcd797cd9211) fix that was merged and released under `2.4.2`
> 
> I think this will work with any version of `uglify-js` that is `>=3.x`

@ghiscoding, would have spend days in this issue, if i haven't found your reply. Thank you.

**Comment by alan-agius4** [comment]
Thanks for reporting this issue. This issue is now obsolete due to changes in the recent releases. Please update to the most recent ng-packagr.

If the problem persists after upgrading, please open a new issue, provide a simple repository reproducing the problem, and describe the difference between the expected and current behavior.
